lib/core/app/app_router.dart

```diff
@@ -101,13 +105,46 @@ class AppRouter {
           ),
         ),
         GoRoute(
-          path: Pages.recommend.path,
-          name: Pages.recommend.name,
-          builder: (context, state) => BlocProvider<RecommendationCubit>(
-            create: (context) => RecommendationCubit(getIt()),
-            child: const RecommendationScreen(),
+          path: Pages.locationSelection.path,
+          name: Pages.locationSelection.name,
+          builder: (context, state) => BlocProvider(
+            create: (context) => TagsCubit(
+              getIt(),
+            )..fetchTags(),
+            child: const LocationSelectionScreen(),
           ),
         ),
+        GoRoute(
+          path: Pages.interestsSelection.path,
+          name: Pages.interestsSelection.name,
+          builder: (context, state) {
+            final tagsState = (state.extra! as TagsStateFetched);
+            return BlocProvider(
+              create: (context) => InterestsCubit(
+                location: tagsState.selectedLocation,
+                interests: tagsState.interests,
+              ),
+              child: const InterestsSelectionScreen(),
+            );
+          },
+        ),
+        GoRoute(
+          path: Pages.organisations.path,
+          name: Pages.organisations.name,
```

Comment:
Can it be a bit more descriptive like recommendedOrganisations ?

Reply:
fixed